src/api/Method.ts

```diff
@@ -1,147 +1,147 @@
+    if (queryParams[key] === true && !params[key]) {
+      console.error(`Missing required query parameter: ${key}`);
+      throw new DimoError({
+        message: `Missing required query parameter: ${key}`,
+        statusCode: 400,
+      });
     }
-    
-    /**
-     * Query Parameters
-     * Extract required queryParams from the resource object
-     */ 
-
-    const queryParams = resource.queryParams || {};
-    // Check if required queryParams match with incoming params
-    for (const key in queryParams) {
-        // We'll fail early if there's a missing required query param from the user
-        if (queryParams[key] === true && !params[key]) {
-            console.error(`Missing required query parameter: ${key}`);
-            throw new DimoError({
-                message: `Missing required query parameter: ${key}`,
-                statusCode: 400
-            });
-        }
-        if (queryParams[key][0] === '$') {
-            const variableKey = queryParams[key].substring(1); // Remove the leading $
-            if (params[variableKey]) {
-                params[key] = params[variableKey]; // Set the value based on the value of the variable key
-            } else {
-                console.error(`Variable key '${variableKey}' not found in params`);
-                throw new DimoError({
-                    message: `Variable key '${variableKey}' not found in params`,
-                    statusCode: 400
-                });
-            }
-        } else if (typeof queryParams[key] === 'string') {
-            params[key] = queryParams[key];
-        }
+    if (queryParams[key][0] === "$") {
+      const variableKey = queryParams[key].substring(1); // Remove the leading $
+      if (params[variableKey]) {
+        params[key] = params[variableKey]; // Set the value based on the value of the variable key
+      } else {
+        console.error(`Variable key '${variableKey}' not found in params`);
+        throw new DimoError({
+          message: `Variable key '${variableKey}' not found in params`,
+          statusCode: 400,
+        });
+      }
+    } else if (typeof queryParams[key] === "string") {
+      params[key] = queryParams[key];
     }
+  }
 
-    /**
-     * URL Parameters
-     * Check for placeholders in the resource path and replace them with values from params
-     */ 
+  /**
+   * URL Parameters
+   * Check for placeholders in the resource path and replace them with values from params
+   */
 
-    let url = baseUrl + resource.path;
+  let url = baseUrl + resource.path;
 
-    const placeholderRegex = /:([\w]+)/g;
-    let match;
-    while ((match = placeholderRegex.exec(resource.path)) !== null) {
-        const key = match[1];
-        const value = params[key];
-        if (value !== undefined) {
-            url = url.replace(match[0], value.toString());
-        }
+  const placeholderRegex = /:([\w]+)/g;
+  let match;
+  while ((match = placeholderRegex.exec(resource.path)) !== null) {
```

Comment:
I believe `exec` just returns either `null` or an array, but I think I would be more comfortable with `while (match = placeholderRegex.exec(resource.path))`